examples/ipam-vpc/main.tf

```diff
@@ -19,17 +20,73 @@ locals {
 # VPC Module
 ################################################################################
 
-module "vpc" {
+# IPv4
+module "vpc_without_ipam" {
   source = "../.."
 
-  name = local.name
+  name = "${local.name}-without-ipam"
+  cidr = "10.0.0.0/16"
+  azs  = local.azs
 
-  private_subnets = cidrsubnets(local.partition[0], 2, 2)
-  public_subnets  = cidrsubnets(local.partition[1], 2, 2)
+  private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets  = ["10.0.11.0/24", "10.0.12.0/24", "10.0.13.0/24"]
 
+  tags = local.tags
+}
+
+module "vpc_ipam_set_cidr" {
+  source = "../.."
+
+  name = "${local.name}-set-cidr"
+
+  use_ipam_pool     = true
   ipv4_ipam_pool_id = aws_vpc_ipam_pool.this.id
-  azs               = ["${local.region}a", "${local.region}b"]
-  cidr              = aws_vpc_ipam_preview_next_cidr.this.cidr
+  cidr              = "10.0.0.0/16"
+  azs               = local.azs
+
+  private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets  = ["10.0.11.0/24", "10.0.12.0/24", "10.0.13.0/24"]
+
+  tags = local.tags
+
+  depends_on = [
+    aws_vpc_ipam_pool_cidr.this
+  ]
+}
+
+module "vpc_ipam_set_netmask" {
+  source = "../.."
+
+  name = "${local.name}-set-netmask"
+
+  use_ipam_pool       = true
+  ipv4_ipam_pool_id   = aws_vpc_ipam_pool.this.id
+  ipv4_netmask_length = 28
+  azs                 = local.azs
+
+  private_subnets = cidrsubnets(local.preview_partition[0], 2, 2, 2)
+  public_subnets  = cidrsubnets(local.preview_partition[1], 2, 2, 2)
+
+  tags = local.tags
+
+  depends_on = [
+    aws_vpc_ipam_pool_cidr.this
+  ]
+}
+
+# IPv6
+module "vpc_ipv6_ipam_set_netmask" {
+  source = "../.."
+
+  name = "${local.name}-ipv6-set-netmask"
+
+  use_ipam_pool       = true
+  ipv6_ipam_pool_id   = aws_vpc_ipam_pool.ipv6.id
+  ipv6_netmask_length = 56
```

Comment:
@drewmullen do you know whats up with this - if I set a netmask length of 56 I get

```
│ Error: error creating EC2 VPC: InvalidParameterValue: The allocation size is too big for the pool.
│ 	status code: 400, request id: be5c7702-a7f7-40ef-a6a8-25a2523a7fb6
│
│   with module.vpc_ipv6_ipam_set_netmask.aws_vpc.this[0],
│   on ../../main.tf line 20, in resource "aws_vpc" "this":
│   20: resource "aws_vpc" "this" {
```

And if I set anything other than a netmask length of 56 I get

```
│ Error: expected ipv6_netmask_length to be one of [56], got 60
│
│   with module.vpc_ipv6_ipam_set_netmask.aws_vpc.this[0],
│   on ../../main.tf line 30, in resource "aws_vpc" "this":
│   30:   ipv6_netmask_length              = var.ipv6_netmask_length
```

Reply:
You have to provision an IPv6 into the pool using `aws_vpc_ipam_pool_cidr` for it to have a prefix to allocate. ipam ipv6 is similar to how byoip worked prior to the service. Heres some [test examples](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/ec2/ipam_byoip_test.go#L211-L242) in the terraform acceptance tests. 

As you can see, those tests dont normally run during TF builds unless the account has a IP it is authorized to assume control of. You only run the tests if you have set variables: `IPAM_BYOIP_IPV6_MESSAGE`, `IPAM_BYOIP_IPV6_SIGNATURE`, `IPAM_BYOIP_IPV6_PROVISIONED_CIDR`

Reply:
I'm dumb, I thought I read `aws_vpc_ipam_pool_cidr` was only valid for IPv4

Reply:
For now I have just commented out the IPv6 bits since they aren't valid in the example unless you have those 3 pieces of information. For some reason, I thought I had read that IPAM would vend you a /56 CIDR from the global pool when requested (similar to how VPC works with assigning an IPv6 CIDR when enabled).

Reply:
You could always just buy a public IPv6 space 🤣 seems worth it to validate the tests

![image](https://user-images.githubusercontent.com/3812162/192273392-97851de2-e8a7-4ad4-a7be-3b150200e047.png)